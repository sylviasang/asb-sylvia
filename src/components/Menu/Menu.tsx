import React from 'react';
import styled from 'styled-components';
import { IconButton } from '@material-ui/core';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';

const Container = styled.div`
    display: flex;
    flex-direction: column;
`;

const NavigationContainer = styled.div`
    display: flex;
    flex-direction: row;
    border-bottom: 1px solid #000000;
`;

const Header = styled.div`
    width: 100%;
    display: flex;
    justify-content: center;
`;

const FormContainer = styled.div`
    display: flex;
    justify-content: center;
    margin-left: 180px;
`;

interface MenuProps { 
  handleClick: any
};

const Menu:React.FC<MenuProps> = (props) => {
  return (
    <Container>
      <NavigationContainer>
          <IconButton name='ArrowBackBotton' data-testid='ArrowBack' onClick={() => props.handleClick()}><ArrowBackIcon fontSize='large'/></IconButton>
          <Header><h1 data-testid='Menu'>Menu</h1></Header>
      </NavigationContainer>
      <FormContainer>
        <h2 style={{marginTop: '20px'}}>This is menu content</h2>
      </FormContainer>
    </Container>
  );
}

export default Menu;
