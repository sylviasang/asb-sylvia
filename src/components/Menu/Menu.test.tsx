import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import Menu from './Menu';

afterEach(() => {
    cleanup();
})

test('render with props', () => {
    const onClick = jest.fn();
    const { container } = render(<Menu handleClick={onClick}/>);
    expect(container.firstChild).toMatchSnapshot();
})

test('render when arrow button click', () => {
    const onClick = jest.fn();
    const { container } = render(<Menu handleClick={onClick}/>);
    fireEvent.click(screen.getByTestId('ArrowBack'));
    expect(onClick).toHaveBeenCalled();
})