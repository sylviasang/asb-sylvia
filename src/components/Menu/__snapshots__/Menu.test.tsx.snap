// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with props 1`] = `
<div
  class="sc-bdfBQB aujAO"
>
  <div
    class="sc-gsTEea hGQBnA"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      data-testid="ArrowBack"
      name="ArrowBackBotton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="sc-dlfnuX jzRvPj"
    >
      <h1
        data-testid="Menu"
      >
        Menu
      </h1>
    </div>
  </div>
  <div
    class="sc-hKgJUU hZthbb"
  >
    <h2
      style="margin-top: 20px;"
    >
      This is menu content
    </h2>
  </div>
</div>
`;
